Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -63,11 +53,570 @@ ptree MarvinMolBase::toPtree() const {
 }
 
 void MarvinMolBase::addSgroupsToPtree(ptree &out) const {
-  for (auto sgroup : sgroups) {
+  for (auto &sgroup : sgroups) {
     out.add_child("molecule", sgroup->toPtree());
   }
 }
 
+bool getCleanDouble(std::string strToParse, double &outDouble) {
```

Comment:
This could almost be a templated method.

Reply:
Done - getCleanNumber